src/Tizen.NUI/src/public/WindowSystem/QuickPanelClient.cs

```diff
@@ -0,0 +1,394 @@
+            {
+                throw new ArgumentNullException((string)"win");
+            }
+
+            _tzsh = tzShell;
+            _tzshWin = win.GetNativeId();
+            _tzshQpClient = Interop.QuickPanelClient.tzshQuickPanelCreateWithType(_tzsh.GetNativeHandle(), (IntPtr)_tzshWin, (int)type);
+            if (_tzshQpClient == (IntPtr)0)
+            {
+                int err = Tizen.Internals.Errors.ErrorFacts.GetLastResult();
+                _tzsh.ErrorCodeThrow(err);
+            }
+        }
+
+        /// <summary>
+        /// Destructor.
+        /// </summary>
+        ~QuickPanelClient()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(bool disposing)
+        {
+            int _res;
+            if (!disposed)
+            {
+                if (_tzshQpClient != (IntPtr)0)
+                {
+                    _res = Interop.QuickPanelClient.tzshQuickPanelDestroy(_tzshQpClient);
+                    _tzsh.ErrorCodeThrow(_res);
+                    _tzshQpClient = (IntPtr)0;
+                }
+                disposed = true;
+            }
+        }
+
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        private void OnVisibleChanged(int type, IntPtr ev_info, IntPtr data)
+        {
+            int _res;
+            IntPtr _state = IntPtr.Zero;
+            StateVisible state;
+
+            _res = Interop.QuickPanelClient.tzshQuickPanelEventVisibleGet(ev_info, ref _state);
+            _tzsh.ErrorCodeThrow(_res);
+
+            state = (StateVisible)_state;
+            _visibleChanged?.Invoke(this, state);
+        }
+
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        private void OnOrientationChanged(int type, IntPtr ev_info, IntPtr data)
+        {
+            int _res;
+            IntPtr _state = IntPtr.Zero;
+            StateOrientation state;
+
+            _res = Interop.QuickPanelClient.tzshQuickPanelEventOrientationGet(ev_info, ref _state);
+            _tzsh.ErrorCodeThrow(_res);
+
+            state = (StateOrientation)_state;
+            _orientationChanged?.Invoke(this, state);
+        }
+
+        /// <summary>
+        /// Gets the visible state of the quickpanel.
+        /// </summary>
+        /// <returns>The visible state of the quickpanel service window.</returns>
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of an invalid operation or no service.</exception>
```

Comment:
Please add `<since_tizen>8</since_tizen>` tag to all public API.

Reply:
I added since_tizen doc to all public API